.github/workflows/gitleaks.yml

```
@@ -0,0 +1,23 @@
+name: gitleaks
+
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 * * *" # run once a day at 4 AM UTC
+
```

need to add permission block
```suggestion
permissions: 
  contents: read
  security-events: write
  actions: read
  issues: write
  pull-requests: write
```